### PR TITLE
fix: dimension min max column only for hgv and trl

### PR DESCRIPTION
**Pull request by m-ujaffer**

### CODE REVIEW

File: src/app/technical-record/dimensions/__snapshots__/dimensions.component.spec.ts.snap [hunk 1]
```diff
@@ -99,32 +99,7 @@ exports[`DimensionsComponent should create view only with populated data 1`] = `
...
-          class="govuk-table__cell"
-        />
-        <td
-          class="govuk-table__cell"
-        />
-        <td
-          class="govuk-table__cell"
-        />
-      </tr>
+
```
Comment (alleypa) [inline src/app/technical-record/dimensions/__snapshots__/dimensions.component.spec.ts.snap:102]: Umer let update the existing `should create view only with populated data` with these 2 tests
- should create view only with HGV or TRL populated data
- should create view only with other vehicle types

File: src/app/utils/testing.utils.ts [hunk 2]
```diff
@@ -397,3 +399,14 @@ function mockAxleSpacing(args?: Partial<AxleSpacing>): AxleSpacing {
 
   return {...mock, ...args};
 }
+
```
Comment (alleypa) [inline src/app/utils/testing.utils.ts:402]: Use this `function mockDimensions(args?: Partial<Dimensions>): Dimensions {
  const mock: Dimensions = {
    length: 100,
    width: 100,
    height: 200
  } as Dimensions;

  return { ...mock, ...args };
}`

compose this `axleSpacing: [mockAxleSpacing()]` in your test if need be. But you dont need it

File: src/app/technical-record/dimensions/dimensions.component.spec.ts [hunk 3]
```diff
@@ -24,24 +26,32 @@ describe('DimensionsComponent', () => {
...
-                                            length: 100,
-                                            width: 100,
-                                            axleSpacing: [
-                                              { axles: '1', value: 100},
-                                              { axles: '2', value: 100},
-                                              { axles: '3', value: 100},
-                                            ]
-                                            }
-                              } as TechRecord;
+        frontAxleToRearAxle: 450,
```
Comment (alleypa) [inline src/app/technical-record/dimensions/dimensions.component.spec.ts:29]: Extract into reusable like this with this `const getTechRecord = (): TechRecord => {
  return {
    frontAxleTo5thWheelMin: 456,
    frontAxleTo5thWheelMax: 660,
    frontAxleTo5thWheelCouplingMin: 100,
    frontAxleTo5thWheelCouplingMax: 150,
    dimensions: TESTING_UTILS.mockDimensions({
      axleSpacing: [
        { axles: '1', value: 100 },
        { axles: '2', value: 100 },
        { axles: '3', value: 100 }
      ]
    })
  } as TechRecord;
};`

File: src/app/technical-record/dimensions/dimensions.component.spec.ts [hunk 4]
```diff
@@ -24,24 +26,32 @@ describe('DimensionsComponent', () => {
...
+            { axles: '3', value: 100 }
+          ]
+        })
+      } as TechRecord;
     component.activeRecord = tecRec;
   });
 
-  it('should create view only with populated data', () => {
+  it('should create view only for hgv with populated data', () => {
+    component.activeRecord.vehicleType = VEHICLE_TYPES.HGV;
```
Comment (alleypa) [inline src/app/technical-record/dimensions/dimensions.component.spec.ts:46]: component.activeRecord = { ...this.getTec(), vehicleType: VEHICLE_TYPES.HGV } as TechRecord;

File: src/app/technical-record/dimensions/dimensions.component.spec.ts [hunk 5]
```diff
@@ -24,24 +26,32 @@ describe('DimensionsComponent', () => {
...
+    fixture.detectChanges();
+
+    expect(component).toBeDefined();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should create view only for psv with populated data', () => {
+    component.activeRecord.vehicleType = VEHICLE_TYPES.PSV;
```
Comment (alleypa) [inline src/app/technical-record/dimensions/dimensions.component.spec.ts:54]: component.activeRecord =` { ...this.getTechRecord (), vehicleType: VEHICLE_TYPES.PSV } as TechRecord;`

File: src/app/technical-record/dimensions/dimensions.component.spec.ts [hunk 6]
```diff
@@ -24,24 +26,32 @@ describe('DimensionsComponent', () => {
...
-  it('should create view only with populated data', () => {
+  it('should create view only for hgv with populated data', () => {
+    component.activeRecord.vehicleType = VEHICLE_TYPES.HGV;
+    fixture.detectChanges();
+
+    expect(component).toBeDefined();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should create view only for psv with populated data', () => {
```
Comment (alleypa) [inline src/app/technical-record/dimensions/dimensions.component.spec.ts:53]: I think this is better `should create view only with other vehicle types` whilst our focus is only on HVG and TRL

File: src/app/technical-record/dimensions/__snapshots__/dimensions.component.spec.ts.snap (hunk 1 above)
Comment (m-ujaffer) [reply src/app/technical-record/dimensions/__snapshots__/dimensions.component.spec.ts.snap:414]: done

File: src/app/utils/testing.utils.ts (hunk 2 above)
Comment (m-ujaffer) [reply src/app/utils/testing.utils.ts:406]: done

File: src/app/technical-record/dimensions/dimensions.component.spec.ts (hunk 5 above)
Comment (m-ujaffer) [reply src/app/technical-record/dimensions/dimensions.component.spec.ts]: done

File: src/app/technical-record/dimensions/dimensions.component.spec.ts (hunk 6 above)
Comment (m-ujaffer) [reply src/app/technical-record/dimensions/dimensions.component.spec.ts]: done

File: src/app/technical-record/dimensions/dimensions.component.spec.ts (hunk 4 above)
Comment (m-ujaffer) [reply src/app/technical-record/dimensions/dimensions.component.spec.ts]: done

File: src/app/technical-record/dimensions/dimensions.component.spec.ts (hunk 3 above)
Comment (m-ujaffer) [reply src/app/technical-record/dimensions/dimensions.component.spec.ts]: done

**Pull request merged**